app/src/main/java/com/geeksville/mesh/ui/NodeDetailsFragment.kt

```diff
@@ -0,0 +1,156 @@
+import androidx.fragment.app.viewModels
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import com.geeksville.mesh.DeviceMetrics
+import com.geeksville.mesh.R
+import com.geeksville.mesh.android.Logging
+import com.geeksville.mesh.model.DebugViewModel
+import com.geeksville.mesh.model.UIViewModel
+import com.geeksville.mesh.ui.components.DataEntry
+import com.geeksville.mesh.ui.components.DeviceMetricsCard
+import com.geeksville.mesh.ui.components.DeviceMetricsChart
+import com.geeksville.mesh.ui.theme.AppTheme
+import dagger.hilt.android.AndroidEntryPoint
+
+internal fun FragmentManager.navigateToNodeDetails(nodeNum: Int? = null) {
+    val nodeDetailsFragment = NodeDetailsFragment().apply {
+        arguments = bundleOf("nodeNum" to nodeNum)
+    }
+    beginTransaction()
+        .replace(R.id.mainActivityLayout, nodeDetailsFragment)
+        .addToBackStack(null)
+        .commit()
+}
+
+@AndroidEntryPoint
+class NodeDetailsFragment : ScreenFragment("NodeDetails"), Logging {
+
+    private val model: UIViewModel by activityViewModels()
+
+    private val debugModel: DebugViewModel by viewModels()
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        val nodeNum = arguments?.getInt("nodeNum")
+
+        return ComposeView(requireContext()).apply {
+            setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
+            setContent {
+                AppTheme {
+                    NodeDetailsScreen(
+                        model = model,
+                        debugModel = debugModel,
+                        nodeNum = nodeNum,
+                        navigateBack = {
+                            parentFragmentManager.popBackStack()
+                        }
+                    )
+                }
+            }
+        }
+    }
+}
+
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+fun NodeDetailsScreen(
+    model: UIViewModel = hiltViewModel(),
+    debugModel: DebugViewModel, // TODO do I need something similar to hiltViewModel()?
+    nodeNum: Int? = null,
+    navigateBack: () -> Unit,
+) {
+    val ourNodeInfo by model.ourNodeInfo.collectAsStateWithLifecycle()
+    val selectedNodeInfo by model.getNodeByNum(nodeNum ?: ourNodeInfo?.num ?: 0)
+        .collectAsStateWithLifecycle()
+
+    val nodeId = selectedNodeInfo?.user?.id
+    val logs by debugModel.meshLog.collectAsStateWithLifecycle()
+    val data = mutableListOf<DataEntry>()
+
+    /* Retrieve only the data belonging to the selected node. */
+    for (log in logs) {
+        log.nodeInfo?.let { nodeInfo ->
+            val currentId = nodeInfo.user.id
+            if (nodeInfo.hasDeviceMetrics() && currentId == nodeId)
+                data.add(DataEntry(log.received_date, DeviceMetrics(nodeInfo.deviceMetrics)))
+        }
+    }
```

Comment:
we want to do all the heavy lifting in the ViewModel, and the UI should be just the consumer for the data.

you can add your own custom ViewModel ("NodeDetailsViewModel"?) using "DebugViewModel" as a baseline and build the `DataEntry` list there.

I added a method to the MeshLog repository to retrive Telemetry for a specific node that you can use: https://github.com/meshtastic/Meshtastic-Android/blob/7e0cfff67bd38f0eaeed05798456c2e8c430fcf2/app/src/main/java/com/geeksville/mesh/database/MeshLogRepository.kt#L36

Reply:
Awesome with that new method I was able to get all the data!

![Screenshot_20240818-225816_Meshtastic](https://github.com/user-attachments/assets/86f7d7b1-78b8-4d72-a9b7-f4b0b7cf7f5c)